examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/NetworkCredentialsParcelable.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package com.google.chip.chiptool
+
+import android.os.Parcelable
+import android.os.Parcel
+import android.os.Parcelable.Creator
+
+/** Class for holding WiFi or Thread credentials, but not both.  */
+class NetworkCredentialsParcelable : Parcelable {
+  var wiFiCredentials: WiFiCredentials?
+    private set
+  var threadCredentials: ThreadCredentials?
+    private set
+
+  private constructor(
+    wifiCredentials: WiFiCredentials?, threadCredentials: ThreadCredentials?
+  ) {
+    wiFiCredentials = wifiCredentials
+    this.threadCredentials = threadCredentials
+  }
+
+  // Begin Parcelable implementation
+  private constructor(`in`: Parcel) {
```

Comment:
is this a typical pattern in kotlin? How about naming it `input` or `parcel` to not have the backticks?

Reply:
same comment on several instances below (although hit does look like https://developer.android.com/reference/android/os/Parcelable has the kotlin example using the backticks ... so feel free to also close the comment as "this is the kotlin way")

Reply:
acked